rest-api/dao/participant_dao.py

```diff
@@ -165,6 +165,14 @@ def validate_participant_id(self, session, participant_id):
     raise_if_withdrawn(participant)
     return participant
 
+  def check_participant_exist(self, participant_id):
```

Comment:
I don't think this quite the way we want to do things... this will do an additional database request in a different session (which doesn't guarantee technically that the participant still exists by the time we get to the insert code.)

We already have validate_participant_reference, which is currently being called during _update_participant_summary here:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/dao/questionnaire_response_dao.py#L169

That is unfortunately *after* we reference participant.participantSummary here:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/dao/questionnaire_response_dao.py#L154

Which is the source of the error.

So what we probably want to do is just have a check above that line that participant (which we fetched for update) is not None, and throw a BadRequestException if it is. (We've been returning NotFound specifically only when trying to retrieve some resource that doesn't exist, rather than returning it for creating a resource on a participant that doesn't exist.) We can then remove the call to validate_participant_reference in the update code I believe.